crates/sargon/src/system/sargon_os/sargon_os.rs

```diff
@@ -151,6 +151,38 @@ impl SargonOS {
         Ok(())
     }
 
+    pub async fn derive_wallet(
+        &self,
+        device_factor_source: DeviceFactorSource,
+        accounts: Accounts,
+    ) -> Result<()> {
+        debug!("Deriving Profile from BDFS");
+
+        let host_id = self.host_id().await?;
+        let host_info = self.host_info().await;
+
+        let profile = Profile::from_device_factor_source(
+            device_factor_source,
+            host_id,
+            host_info,
+            Some(accounts),
+        );
+
+        self.secure_storage.save_profile(&profile).await?;
+        let profile_state = ProfileState::Loaded(profile);
+        self.profile_state_holder
+            .replace_profile_state_with(profile_state.clone())?;
+
+        self.clients.profile_state_change.emit(profile_state).await;
```

Comment:
`set_profile` should probably be updated to allow changing to Loaded state with a profile, and then it could be used directly.

Reply:
I've made the change here https://github.com/radixdlt/sargon/pull/212/commits/3f32ddb0f28e740af2f04865576f97a199e85a3c. Decided to also use `set_profile` in `new_wallet` too. Let me know your thoughts.